gui/editor/editor.cpp

```diff
@@ -8264,6 +8271,34 @@ std::pair<int,int> NumeReEditor::getCurrentContext(int line)
 }
 
 
+/////////////////////////////////////////////////
+/// \brief This helper function finds the line
+/// to start code execution for the run from
+/// line command. This includes checking for
+/// continued lines and loops.
+///
+/// \param line int
+/// \return int
+///
+/////////////////////////////////////////////////
+int NumeReEditor::getStartLine(int line)
+{
+    // Only check for scripts and do not do further checks for line 0
+    if (m_fileType != FILE_NSCR || line == 0)
+        return 0;
+
+    // If in a loop, find start of loop
+    while (GetFoldParent(line - 1) != wxNOT_FOUND)
+        line = GetFoldParent(line - 1) + 1;

```

Comment:
Yes, this is because `GetFoldParent` works with 0 based indexing and this function works with 1 based. So these two are the transformations into and from the 0 indexing.